### PR TITLE
OCC-25: Implement GDPR compliant deletion

### PR DESCRIPTION
**Pull request by nlubisch**
### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [ ] I have squashed any insignificant commits
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code

### CODE REVIEW

**Comment by MacGyver-NRW** [comment]
Approved working functionality by remote view. **Great job, pal!**
Delete and modify all personal information to follow GDPR.

Causes: new option in admin view
- based on personal access, enter "admin" menu in OC
- select a user who want's to be deleted
- verify and confirm user
- check summary of recordings
- be sure to set "Delete" AND "GDPR" checkbox to execute this process
- success notice will show up

**Review by MacGyver-NRW — APPROVED** [review]
Live tested on local developer machine | remote view

File: htdocs/lib2/logic/user.class.php [hunk 1]
```diff
@@ -932,9 +932,9 @@ public function canDisable()
         global $login;
         $login->verify();
 
-        if ($login->userid != $this->nUserId && ($login->admin & ADMIN_USER) != ADMIN_USER) {
-            return false;
-        }
+//        if ($login->userid != $this->nUserId && ($login->admin & ADMIN_USER) != ADMIN_USER) {
+//            return false;
+//        }
```
Comment (teiling88) [inline htdocs/lib2/logic/user.class.php]: ???

File: htdocs/lib2/logic/user.class.php [hunk 2]
```diff
@@ -1032,9 +1032,9 @@ public function canDisableDueLicense()
...
```
Comment (teiling88) [inline htdocs/lib2/logic/user.class.php]: ???

File: htdocs/lib2/logic/user.class.php [hunk 3]
```diff
@@ -1483,16 +1483,7 @@ public function canDelete()
...
-            return false;
-        }
-
-        return
-            sql_value("SELECT COUNT(*) FROM `caches` WHERE `user_id`='&1'", 0, $this->nUserId)
-            + sql_value("SELECT COUNT(*) FROM `cache_logs` WHERE `user_id`='&1'", 0, $this->nUserId)
-            + sql_value("SELECT COUNT(*) FROM `cache_logs_archived` WHERE `user_id`='&1'", 0, $this->nUserId)
-            + sql_value("SELECT COUNT(*) FROM `cache_reports` WHERE `userid`='&1'", 0, $this->nUserId)
-            == 0;
+        return !($login->userid != $this->nUserId && ($login->admin & ADMIN_USER) != ADMIN_USER);
```
Comment (teiling88) [inline htdocs/lib2/logic/user.class.php]: Das ist doch nicht das gleiche wie vorher? Auch wenn Du es jetzt einfach negiert hast. In dem zweiten return kann ja true und false bei rum kommen.

**Review by teiling88 — CHANGES_REQUESTED** [review]

**Review by teiling88 — APPROVED** [review]

**Pull request merged**